example/weather/services/tester/run_tests.go

```diff
@@ -162,22 +201,50 @@ func (svc *Service) runTests(ctx context.Context, p *gentester.TesterPayload, te
 
 	// Run the tests that need to be run and add the results to the testCollection.Results array
 	if runSynchronously {
+		// RunSynchronously is used for test collections that need to be run one after in order
+		// to avoid single resource contention between tests if they are run in parallel. An
+		// example of this is tests that rely on the same cloud resource, such as a spreadsheet,
+		// as part of their test functionality.
+		//
+		// testName is passed to recoverFromTestPanic as a function so that, via a closure, its
+		// name can be set before the test is run but after the defer of recoverFromTestPanic is
+		// declared. This is done because the test name is not accessible from within the test
+		// function itself where it is set.
 		log.Infof(ctx, "RUNNING TESTS SYNCHRONOUSLY")
-		for n, test := range testsToRun {
-			log.Infof(ctx, "RUNNING TEST [%v]", n)
-			test(ctx, testCollection)
-		}
+		wg := sync.WaitGroup{}
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			testName := ""
+			testNameFunc := func() string {
+				return testName
+			}
+			defer recoverFromTestPanic(ctx, testNameFunc, testCollection)
+			for _, test := range testsToRun {
+				testName = getTestName(test)
+				log.Infof(ctx, "RUNNING TEST [%v]", testName)
+				test(ctx, testCollection)
+			}
+		}()
+		wg.Wait()
 	} else {
+		// if not run synchronously, run the tests in parallel and assumed not to have resource
+		// contention
 		log.Infof(ctx, "RUNNING TESTS IN PARALLEL")
 		wg := sync.WaitGroup{}
-		for n, test := range testsToRun {
+		for _, test := range testsToRun {
 			wg.Add(1)
-			go func(f func(context.Context, *TestCollection), testName string) {
+			go func(f func(context.Context, *TestCollection)) {
 				defer wg.Done()
-				defer recoverFromTestPanic(ctx, testName, testCollection)
+				testName := ""
+				testNameFunc := func() string {
+					return testName
+				}
+				defer recoverFromTestPanic(ctx, testNameFunc, testCollection)
+				testName = getTestName(f)
 				log.Infof(ctx, "RUNNING TEST [%v]", testName)
 				f(ctx, testCollection)
-			}(test, n)
```

Comment:
The previous version was much better :)

Reply:
@raphael not sure I follow, I took the testName out because I am using getTestName and didn't need to pass in testName anymore to the goroutine. If I figure out how to get testName without reflection this may go back though.